frontend/src/app/components/header/header.component.ts

```diff
@@ -8,14 +10,18 @@ import { Router, NavigationEnd } from '@angular/router';
 })
 export class HeaderComponent implements OnInit {
 
-  constructor(private router : Router) { }
+  public isLoggedIn : Observable<boolean>;
+
+  constructor(private router : Router, private sessionService : SessionService) { 
+    this.isLoggedIn = sessionService.isLoggedIn();
+  }
 
   ngOnInit() {
   }
 
 
   public logout(){
-    localStorage.clear();
+    this.sessionService.logout();
```

Comment:
there might be a case when the logout logic fails and there is no fallback, the redirect happens anyways
there are two solutions for now:
1. wrap the return as a Promise and then call the navigate

```suggestion
    this.sessionService.logout().then(() => {
         this.router.navigateByUrl('');
    });
```